### ISSUE TITLE
Pasted images are not visible in Ticket preview and opened ticket

### ISSUE THREAD

**Issue by Commander67** [a-ono/redmine_ckeditor#310]
When I paste an image in the editor I can see it well and have WYSIWIG-view. 
But it is not visible in the preview tab and not visible after saving.
Additionally I can't refer to an uploaded image using redmines !image.png!.
Any ideas?

**Comment by ChrisUHZ** [comment]
If you paste images directly they will be encoded and placed into html code as base64 "data" Protocol.
You have to enable this protocol in the config file from CKEditor!
(see example file in the plugin directory of your redmine installation/redmine_ckeditor/config/ckeditor.yml.example and edit the file with your needs and copy the file into redmine installation/config/ckeditor.yml)

allowedProtocols:
.....
  - urn
  - webcal
  - xmpp
  **- data**

restart redmine/apache and you will be fine :)

**Comment by MasyGreen** [comment]
is solution dose not work( Rm4.1

**Comment by RobertoCesar** [comment]
The following worked for me:
  Add the following line in the end of the `plugins/redmine_ckeditor/init.rb` file:
`(Loofah::VERSION >= "2.3.0" ? Loofah::HTML5::SafeList : Loofah::HTML5::WhiteList)::ALLOWED_PROTOCOLS.add('data')`
  Restart redmine

**Comment by veshus** [comment]
Yeah, config options don't work, but @RobertoCesar patch is OK. Is it possible to add it to the next release?

**Comment by adrianobr** [comment]
Do you need add the plugins: 'pastebase64' and 'base64image'
In config.js add:
config.extraPlugins = 'pastebase64';
config.extraPlugins = 'base64image';

And init.rb add:
(Loofah::VERSION >= "2.3.0" ? Loofah::HTML5::SafeList : Loofah::HTML5::WhiteList)::ALLOWED_PROTOCOLS.add('data')

Working perfect!

**Comment by popy2k14** [comment]
Hi guys, pasting is working but just on edit tickets and not on new tickets.

Does pasting work for you on new tickets?
